### PR TITLE
More impls for struct and union references

### LINKED ISSUE #111: [Rust] Infallible conversion for struct references
Currently we add a `TryFrom` struct references for converting them to their owned versions.

For structs that can be converted infallibly, we should add a `From` implementation instead.

### PR DESCRIPTION
**Pull request by TethysSvensson**
Do some smarter analysis to figure out when struct and union references
should implement more traits like `Into<OwnedVersion>`, `Ord` or `Hash`.

This fixes #111 (and quite a bit more than that).

**Checklist**
- [x] Updated CHANGELOG.md with relevant changes
- [x] Added tests for any new/fixed functionality
- [x] Added/updated documentation for new/changed code
- [x] Checked that README.md still makes sense (and updated it if necessary)

### CODE REVIEW

**Review by kristoff3r — APPROVED** [review]

File: planus-cli/src/codegen/templates/rust/union.template [hunk 1]
```diff
@@ -58,6 +58,139 @@ impl ::core::convert::From<{{info.ref_name}}> for {{info.owned_name}} {
...
+        {% if variants.len() > 1 %}
+            fn discriminant(value: &{{info.ref_name}}<'_>) -> u8 {
+                match value {
+                {% for variant in variants -%}
+                    {{info.ref_name}}::{{ variant.enum_name }}(_) => {{loop.index}},
+                {% endfor %}
+                }
+            }
+            let self_discriminant = discriminant(self);
+            let other_discriminant = discriminant(self);
```
[the remark this answers is not in the document]
Comment (TethysSvensson) [reply planus-cli/src/codegen/templates/rust/union.template:173]: How so?

[the remark this answers is not in the document]
Comment (TethysSvensson) [reply planus-cli/src/codegen/templates/rust/union.template:173]: Ahhh, double self :sweat_smile:

[the remark this answers is not in the document]
Comment (TethysSvensson) [reply planus-cli/src/codegen/templates/rust/union.template:173]: I will add a test and fix it.

[the remark this answers is not in the document]
Comment (TethysSvensson) [reply planus-cli/src/codegen/templates/rust/union.template:173]: After talking to @nicklaswj, I realized I was being stupid. Union references are normal rust enums in our implementation. Everything we need can be implemented using derive.

**Pull request merged**